cgp-utils_config.md

```diff
@@ -0,0 +1,114 @@
+# Configuration
+
+The [cpg-utils library](https://github.com/populationgenomics/cpg-utils) ([pypi](https://pypi.org/project/cpg-utils/)) contains a streamlined config management tool. This config management is used by the seqr-loader pipeline, but is useful in projects at any scale.
+
+This configuration tool uses one or more `TOML` files, and creates a dictionary of key-value attributes which can be accessed at any point, without explicitly passing a configuration object.
+
+## TOML
+
+[Tom's Obvious, Minimal Language](https://toml.io/en/) is a config file format designed to be easily human readable and writeable, with clear data structures. Sections are delineated using bracketed headings, and key-value pairs are defined using `=` syntax, e.g. :
+
+```toml
+global_key = "value"
+
+[heading_1]
+name = "Luke Skywalker"
+age = 53
+
+[heading_1.subheading]
+occupation = ["Jedi", "Hermit", "Force Ghost"]
+```
+
+will be digested into the simple dictionary:
+
+```python
+{
+    'global_key': 'value',
+    'heading_1': {
+        'name': 'Luke Skywalker',
+        'age': 53,
+        'subheading': {
+            'occupation': ["Jedi", "Hermit", "Force Ghost"]
+        }
+    }
+}
+```
+
+## Using cpg-utils.config
+
+To use the `cpg-utils.config` functions, import these two key methods into any script:
+
+```python
+from cpg_utils.config import get_config, set_config_paths
+```
+
+Configuration is loaded lazily, so config files are loaded into memory only when we try to retrieve values from them. This reduces start-up overhead for scripts using this config approach, but can result in late failures if files with invalid content are specified.
+
+### Setting configuration content
```

Comment:
If we are committing to analysis-runner, we don't need to worry about `CPG_CONFIG_PATH` and `set_config_paths`, but instead just pass configs to analysis-runner with multiple `--config` options, and use `copy_common_env` if we want to pass a config down a Batch job.

Reply:
It's still valuable though, in terms of describing how it works under the hood. So maybe move this section under a "development""in depth"/"under the hood" sub-topic?

Reply:
relevant to this comment and the one above, MSFT are interested in separating AIP from dependence on analysis-runner. This guide was going through the mechanics of config.py, with a section at the end addressing how analysis-runner will handle most of these operations for you

Reply:
Maybe it would be helpful to split this into some sort of every-day usage, particularly within CPG, and "advanced" use cases like yours where you can't rely on the analysis-runner? I'd agree with Vlad that otherwise this might be a little confusing for our internal use cases, where you'd never need to call `set_config_paths` explicitly.